Assets/Scripts/UserInterface/anchorSpawner.cs

```diff
@@ -65,33 +65,26 @@ internal virtual void spawnItem(Transform spawnPoint)
 			{
 				prevItem.transform.localScale /= anchorScale;
 			}
-			Transform newObj = null;
-
-			switch (ConstructableObject){
-				//this is a template, please complete the details.
-				case ConstructableObjType.interactablePoint:
-					newObj = InteractablePoint.Constructor().transform;
-					break;
-				default:
-
-				if (item != null)
-				{
-					newObj = Instantiate(item, spawnPoint.transform.position, spawnPoint.rotation).transform;
-				}
-				else
-				{
-					Debug.Log("item is not set. please set item or fix pool. Object: "+gameObject.name);
-				}
-				break;
-
+			Transform newObj;
+			if (!pointSpawner)
+			{
+				newObj = Instantiate(item, transform.position, Quaternion.identity).transform;
+			}
+			else
+			{
+				newObj = GeoObjConstruction.iPoint(HW_GeoSolver.ins.systemPosition(transform.position)).transform;
 			}
 			if(newObj == null)
 			{
 				Debug.LogError("No object spawning.");
 				return;
 			}
 			newObj.transform.localScale *= anchorScale;
-			newObj.GetComponent<AnchorableBehaviour>().TryAttachToNearestAnchor();
+			newObj.GetComponent<AnchorableBehaviour>().anchor = thisAnchor;
+			if (!newObj.GetComponent<AnchorableBehaviour>().TryAttach(true))
+			{
+				Debug.Log("Didnt attach to anchor. Whyyy");
```

Comment:
Informal or unneeded log. Consider commenting out or making it more concise as to why you are logging it.